services/user-management/src/main/java/com/foodie/user/controllers/RoleController.java

```diff
@@ -0,0 +1,105 @@
+package com.foodie.user.controllers;
+
+import org.springframework.data.domain.Page;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import com.foodie.enums.ResponseKeys;
+import com.foodie.user.model.Role;
+import com.foodie.user.model.service.RoleService;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+@Slf4j
+@RestController
+@CrossOrigin
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/users/roles")
+public class RoleController {
+
+    private final RoleService roleService;
+
+    @PostMapping(produces = "application/json", consumes = "application/json")
+    public ResponseEntity<Map<String, ?>> createRole(@Valid @RequestBody Role role) {
+
+        log.info("Creating role: {}", role);
+
+        try {
+            Role createdRole = roleService.save(role);
+            return ResponseEntity.status(HttpStatus.CREATED)
+                    .body(Collections.singletonMap(ResponseKeys.data.toString(), createdRole));
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(Collections.singletonMap(ResponseKeys.error.toString(), "Unable to process your request"));
+        }
+    }
+
+    @GetMapping(path = "/{id}", produces = "application/json")
+    public ResponseEntity<Map<String, ?>> getRoleById(@Valid @PathVariable("id") UUID id) {
+
+        log.info("Fetching role with ID: {}", id);
+
+        try {
+            Optional<Role> role = roleService.findByIdAndDeletedAtIsNull(id);
+
+            return role.<ResponseEntity<Map<String, ?>>>map(
+                    value -> ResponseEntity.ok().body(Collections.singletonMap(ResponseKeys.data.toString(), value)))
+                    .orElseGet(() -> ResponseEntity.status(HttpStatus.NOT_FOUND)
+                            .body(Collections.singletonMap(ResponseKeys.data.toString(), "Role not found")));
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
```

Comment:
You should avoid returning this HTTP status to users as much as possible

Reply:
Yes, you should always return something that provides user friendly response to the caller and shows the other developers that you're handling every known contingencies.